runloop.c

```diff
@@ -1979,12 +1979,17 @@ bool runloop_environment_cb(unsigned cmd, void *data)
 
                   strlcpy(tmp_path, fullpath, sizeof(tmp_path));
                   path_basedir(tmp_path);
+
+                  /* Removes trailing slash */
```

Comment:
OK, you can do one of two possible improvements here:
* (Solution 1) Either make it so that you only do strlen(tmp_path) once, having it do it two times like this is wasteful
* (Solution 2) (Preferred but you will have to see how achievable this is) Not having to use the strlen at all and being able to figure out the size some other way beforehand. But this might not be achievable in which case you can go for solution 1.

Reply:
Couldn't find a way without strlen so I went for solution 1, hope this is OK.